units/provisioning_setup_host.go

```diff
@@ -182,6 +195,68 @@ func (j *setupHostJob) setupHost(ctx context.Context, h *host.Host, settings *ev
 	return nil
 }
 
+func (j *setupHostJob) attachVolume(ctx context.Context) error {
+	mgrOpts, err := cloud.GetManagerOptions(j.host.Distro)
+	if err != nil {
+		return errors.Wrapf(err, "can't get ManagerOpts for '%s'", j.host.Id)
+	}
+	cloudMgr, err := cloud.GetManager(ctx, j.env, mgrOpts)
+	if err != nil {
+		return errors.Wrapf(err,
+			"failed to get cloud manager for host %s with provider %s",
+			j.host.Id, j.host.Provider)
+	}
+
+	// create the volume
+	volume, err := cloudMgr.CreateVolume(ctx, &host.Volume{
+		Size:             j.host.HomeVolumeGB,
+		AvailabilityZone: j.host.Zone,
+		CreatedBy:        j.host.StartedBy,
+		Type:             evergreen.DefaultEBSType,
+	})
+	if err != nil {
+		return errors.Wrapf(err, "can't create a new volume for host '%s'", j.host.Id)
+	}
+
+	// attach to the host
+	attachment := host.VolumeAttachment{VolumeID: volume.ID}
+	if err = cloudMgr.AttachVolume(ctx, j.host, &attachment); err != nil {
+		return errors.Wrapf(err, "can't attach volume '%s' to host '%s'", volume.ID, j.host.Id)
+	}
+
+	if err = j.host.SetHomeVolumeID(volume.ID); err != nil {
+		return errors.Wrapf(err, "can't set home volume ID for host '%s'", j.host.Id)
+	}
+
+	// run the distro's mount script
+	mountCommand := strings.Replace(j.host.Distro.MountCommand, evergreen.DeviceNamePlaceholder, attachment.DeviceName, -1)
+	if j.host.JasperCommunication() {
```

Comment:
is it reasonable to only support jasper communication and error for other cases?

Reply:
done